homeassistant/components/freebox/alarm_control_panel.py

```diff
@@ -0,0 +1,170 @@
+        )
+
+        self.set_state(STATE_IDLE)
+        self.add_features(self._router.home_devices[self._id])
+        self.update_node(self._router.home_devices[self._id])
+
+    @property
+    def state(self) -> str:
+        """Return the state."""
+        return self._state
+
+    async def async_alarm_disarm(self, code: str | None = None) -> None:
+        """Send disarm command."""
+        if await self.set_home_endpoint_value(self._command_off):
+            self.set_state(STATE_ALARM_DISARMED)
+            self.async_write_ha_state()
+
+    async def async_alarm_arm_away(self, code: str | None = None) -> None:
+        """Send arm away command."""
+        if await self.set_home_endpoint_value(self._command_alarm1):
+            self.set_state(STATE_ALARM_ARMING)
+            self.async_write_ha_state()
+
+    async def async_alarm_arm_home(self, code: str | None = None) -> None:
+        """Send arm home command."""
+        if await self.set_home_endpoint_value(self._command_alarm2):
+            self.set_state(STATE_ALARM_ARMING)
+            self.async_write_ha_state()
+
+    async def async_alarm_trigger(self, code: str | None = None) -> None:
+        """Send alarm trigger command."""
+        if await self.set_home_endpoint_value(self._command_trigger):
+            self.set_state(STATE_ALARM_TRIGGERED)
+            self.async_write_ha_state()
+
+    async def async_update_signal(self):
+        """Update signal."""
+        state = await self.get_home_endpoint_value(self._command_state)
+        if state is not None:
+            self.set_state(state)
+            self.update_node(self._router.home_devices[self._id])
+            self.async_write_ha_state()
+
+    def add_features(self, node: dict[str, Any]) -> None:
+        """Add alarm features."""
+        # Search if Alarm2
+        has_alarm2 = False
+        for nodeid, local_node in self._router.home_devices.items():
+            if nodeid == local_node["id"]:
+                alarm2 = next(
+                    filter(
+                        lambda x: (x["name"] == "alarm2" and x["ep_type"] == "signal"),
+                        local_node["show_endpoints"],
+                    ),
+                    None,
+                )
+                if alarm2:
+                    has_alarm2 = alarm2["value"]
+                    break
+
+        if has_alarm2:
+            self._attr_supported_features = (
+                AlarmControlPanelEntityFeature.ARM_AWAY
+                | AlarmControlPanelEntityFeature.ARM_HOME
+            )
+
+        else:
+            self._attr_supported_features = AlarmControlPanelEntityFeature.ARM_AWAY
+
+    def update_node(self, node: dict[str, Any]) -> None:
+        """Update the alarm."""
+        # Parse all endpoints values
+        for endpoint in filter(
+            lambda x: (x["ep_type"] == "signal"), node["show_endpoints"]
+        ):
+            self._attr_extra_state_attributes[endpoint["name"]] = endpoint["value"]
```

Comment:
As sent by Discord, this is showing the alarm code, wich is a security issue

Extra attrs may also not be interesting as already in sensors + for formatted to be used by the user